addons/html_editor/static/src/main/chatgpt/chatgpt_plugin.js

```diff
@@ -129,15 +128,67 @@ export class ChatGPTPlugin extends Plugin {
-        } else {
+        if (params.language) {
             const originalText = selection.textContent() || "";
             this.dependencies.dialog.addDialog(
-                params.language ? ChatGPTTranslateDialog : ChatGPTAlternativesDialog,
+                ChatGPTTranslateDialog,
                 { ...dialogParams, originalText, sanitize }
             );
-        }
+        } else {
+            let recordModel, recordName, recordId, callerId, placeholderPrompt, callerComp, textSelection;
+            // fetch record information that we need for the channel creation
+            const { resModel, resId, data, id } = this.config.getRecordInfo();
+            if (selection.isCollapsed) {
+                if (resModel !== "mail.compose.message") {
+                    const resultName = await this.services.orm.read(resModel, [ resId ], [ 'name' ]);
+                    recordName = resultName[0]['name'];
+                    recordModel = resModel;
+                    recordId = callerId = resId;
+                    callerComp = "html_field_record";
+                    placeholderPrompt = "";
+                    // set insertButtonCaller flag to the records id
+                    this.services['mail.store']["mail.message"].insertButtonCaller = resId;
+                } else {
+                    recordName = data.record_name;
+                    recordModel = data.model;
+                    recordId = Number(data.res_ids.slice(1,-1));
+                    callerComp = "html_field_composer";
+                    callerId = id;
+                    placeholderPrompt = 'Write a follow up answer';
+                }
+            } else {
+                callerComp = "html_field_text_select";
+                recordName = "Text Refine"
+                placeholderPrompt = "Rewrite";
+                textSelection = selection.textContent();
+                callerId = resId;
+                // set insertButtonCaller flag to the records id
+                this.services['mail.store']["mail.message"].insertButtonCaller = resId;
+            }
+            // create the discuss channel used for talking with the ai
+            const ai_channel_id = await this.services.orm.call(
+                'discuss.channel',
+                'create_ai_composer_channel',
+                [ 
+                    callerComp,
+                    recordName,
+                    recordModel,
+                    recordId,
+                    textSelection,
+                ], 
+            );
+            // create and open the thread for the discuss channel
+            const thread = await this.services['mail.store'].Thread.getOrFetch({
+                model: "discuss.channel",
+                id: Number(ai_channel_id), 
+            });
+            thread.open({ 
+                focus: true, 
+                insert: dialogParams.insert,
+                chatCaller: callerId,
+                composerText: placeholderPrompt,
```

Comment:
`composerText` is the value to set in the composer initially? You could just do `thread.composer.text = placeholderPrompt` and call it a day. Much much simpler than the shenanigans of `open()` param with `env` and custom component code to adjust the value.